pwa-shell/src/ui/common/SwipeContentNavigator.js

```diff
@@ -0,0 +1,92 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Gesture } from 'react-with-gesture';
+import { Redirect } from 'react-router-dom';
+
+class SwipeContentNavigator extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
what about setting state outside the constructor? I find it a little cleaner and easier to read (see also "directly inside the class" in https://daveceddia.com/where-initialize-state-react/ )